lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -199,16 +199,25 @@ int modem_key_mgmt_cmp(nrf_sec_tag_t sec_tag,
 		return err;
 	}
 
-	p = scratch_buf;
+	begin = scratch_buf;
 	for (size_t i = 0; i < 3; i++) {
-		p = strchr(p, '\"');
-		if (!p) {
+		begin = strchr(begin, '\"');
+		if (!begin) {
 			return -ENOENT;
 		}
-		p++;
+		begin++;
+	}
+
+	end = strchr(begin, '\"');
+	if (!end) {
+		return -ENOENT;
+	}
+
+	if (end - begin != len) {
+		return -ENOMEM;
 	}
```

Comment:
The check is correct, but the error is wrong. This function return 0 on match, 1 on mismatch (similar to `memcmp` semantics), or a negative value on errors. So in this case, where the length of the credential in modem does not match the length of the credential the application is checking against, we should return 1.